test/built-ins/TypedArray/prototype/buffer/detached-buffer.js

```diff
@@ -0,0 +1,21 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-get-%typedarray%.prototype.buffer
+description: The getter method does not throw with a detached buffer
+info: >
+  22.2.3.1 get %TypedArray%.prototype.buffer
+
+  ...
+  4. Let buffer be the value of O's [[ViewedArrayBuffer]] internal slot.
+  5. Return buffer.
+includes: [testTypedArray.js, detachArrayBuffer.js]
+---*/
+
+var buffer = new ArrayBuffer();
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA(buffer);
+  $DETACHBUFFER(sample);
```

Comment:
Also, it might be a good idea to create a new buffer for each iteration. Implementations might define guards against detaching an already-detached buffer (es6draft does this). It's tricky because we can't account for all behaviors of a non-standard function, but it seems like this is a likely error we could avoid easily.